schema/schema.go

```diff
@@ -0,0 +1,404 @@
+			schema := Schema{
+				Type: "object",
+				Properties: map[string]*Schema{
+					"*": &valueSchema,
+				},
+			}
+			return schema
+		}
+		return Schema{Type: "object"}
+	case reflect.Struct:
+		schema := Schema{
+			Type:       "object",
+			Properties: make(map[string]*Schema),
+		}
+		for i := range t.NumField() {
+			field := t.Field(i)
+
+			if !field.IsExported() {
+				continue
+			}
+
+			jsonTag := field.Tag.Get("json")
+			if jsonTag == "-" {
+				continue
+			}
+
+			fieldName := field.Name
+			required := true
+
+			if jsonTag != "" {
+				parts := strings.Split(jsonTag, ",")
+				if parts[0] != "" {
+					fieldName = parts[0]
+				}
+
+				if slices.Contains(parts[1:], "omitempty") {
+					required = false
+				}
+			} else {
+				fieldName = toSnakeCase(fieldName)
+			}
+
+			fieldSchema := generateSchemaRecursive(field.Type, visited)
+
+			if desc := field.Tag.Get("description"); desc != "" {
+				fieldSchema.Description = desc
+			}
+
+			if enumTag := field.Tag.Get("enum"); enumTag != "" {
+				enumValues := strings.Split(enumTag, ",")
+				fieldSchema.Enum = make([]any, len(enumValues))
+				for i, v := range enumValues {
+					fieldSchema.Enum[i] = strings.TrimSpace(v)
+				}
+			}
+
+			schema.Properties[fieldName] = &fieldSchema
+
+			if required {
+				schema.Required = append(schema.Required, fieldName)
+			}
+		}
+
+		return schema
+	case reflect.Interface:
+		return Schema{Type: "object"}
+	default:
+		return Schema{Type: "object"}
+	}
+}
+
+// ToMap converts a Schema to a map representation suitable for JSON Schema.
+func ToMap(schema Schema) map[string]any {
+	result := make(map[string]any)
+
+	if schema.Type != "" {
+		result["type"] = schema.Type
+	}
+
+	if schema.Description != "" {
+		result["description"] = schema.Description
+	}
+
+	if len(schema.Enum) > 0 {
+		result["enum"] = schema.Enum
+	}
+
+	if schema.Format != "" {
+		result["format"] = schema.Format
+	}
+
+	if schema.Minimum != nil {
+		result["minimum"] = *schema.Minimum
+	}
+
+	if schema.Maximum != nil {
+		result["maximum"] = *schema.Maximum
+	}
+
+	if schema.MinLength != nil {
+		result["minLength"] = *schema.MinLength
+	}
+
+	if schema.MaxLength != nil {
+		result["maxLength"] = *schema.MaxLength
+	}
+
+	if schema.Properties != nil {
+		props := make(map[string]any)
+		for name, propSchema := range schema.Properties {
+			props[name] = ToMap(*propSchema)
+		}
+		result["properties"] = props
+	}
+
+	if len(schema.Required) > 0 {
+		result["required"] = schema.Required
+	}
+
+	if schema.Items != nil {
+		itemsMap := ToMap(*schema.Items)
+		// Ensure type is always set for items, even if it was blank for llama.cpp compatibility
+		if _, hasType := itemsMap["type"]; !hasType && schema.Items.Type == "" {
+			// If type was omitted (useBlankType), restore it for API compatibility
+			if len(schema.Items.Properties) > 0 {
+				itemsMap["type"] = "object"
+			}
+		}
+		result["items"] = itemsMap
+	}
```

Comment:
@kujtimiihoxha Please, consider reviewing the comments to ensure this is as clear as possible.

For example, `useBlankType` is mentioned but that function was actually deleted.